core/domain/skill_domain.py

```diff
@@ -1798,7 +1798,8 @@ class TopicAssignment:
     """
 
     def __init__(
-            self, topic_id, topic_name, topic_version, subtopic_id):
+            self, topic_id, topic_name, topic_version, subtopic_id,
+            skill_ids_for_diagnostic_test):
```

Comment:
`skill_ids_for_siganostic_test` is added as an extra field here because on deleting a question from a skill we have to check two conditions (a). Whether that skill is assigned for the diagnostic test, and (b). Whether that skill contains <= 2 questions.

If both of the conditions are true then a question should be allowed for deletion.
Now for checking condition (a), we have to fetch all the topics which contain that particular skill for the diagnostic test, hence this extra field is added here.

Reply:
I don't think you should do this; what happens if some other method changes that field on the Topic object? This domain object would not get updated, right?

Probably instead, in the relevant methods for deletion etc., you'll need to make the appropriate queries rather than relying on this being up-to-date.

Reply:
But I think you shouldn't base this on the current code that's there, you should also make it robust to future code that's written, and thus my concern still applies. Is it not possible to keep the Topic object/model as the single source of truth?

Reply:
Removed.